app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt

```diff
@@ -122,4 +125,65 @@ class SettingsBasicsTest {
             verifyMenuItemsAreDisabled()
         }
     }
+
+    @SmokeTest
+    @Test
+    fun verifyAddressAutofillTest() {
+        val addressFormPage =
+            TestAssetHelper.getAddressFormAsset(mockWebServer)
+
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openAutofillSubMenu {
+            clickAddAddressButton()
+            fillAndSaveAddress(
+                "Mozilla",
+                "Fenix",
+                "Firefox",
+                "Harrison Street",
+                "San Francisco",
+                "94105",
+                "0123",
+                "mozilla@mozilla.com"
+            )
+        }.goBack {
+        }.goBack {
+        }
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(addressFormPage.url) {
+            clickStreetAddressTextBox()
+            clickSelectAddressButton()
+            clickAddressSuggestion("Harrison Street")
+            verifyAutofilledAddress("Harrison Street")
+        }
+    }
+
+    @SmokeTest
+    @Test
+    fun deleteSavedAddressTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openAutofillSubMenu {
+            clickAddAddressButton()
+            fillAndSaveAddress(
+                "Mozilla",
+                "Fenix",
+                "Firefox",
+                "Harrison Street",
+                "San Francisco",
+                "94105",
+                "0123",
+                "mozilla@mozilla.com"
+            )
```

Comment:
I found https://github.com/mozilla-mobile/fenix/blob/main/app/src/test/java/org/mozilla/fenix/settings/address/interactor/DefaultAddressEditorInteractorTest.kt#L33 we can borrow some of the same inputs e.g, `foo@bar.com` and the `555` phone, default to `US` country

Reply:
Done !